### ISSUE TITLE
Publishing latest image is broken

### ISSUE THREAD

**Issue by pavolloffay** [jaegertracing/jaeger#1952]
Publishing latest image on merge to master is broken https://hub.docker.com/repository/docker/jaegertracing/jaeger-collector

Also we do not publish the image per commit to snapshot repository https://hub.docker.com/repository/docker/jaegertracing/jaeger-collector-snapshot

**Comment by pavolloffay** [comment]
It was broken probably here https://github.com/jaegertracing/jaeger/pull/1909/files#diff-aeee4566b9eb012ce98bd3578f7740a3L26

**Comment by yurishkuro** [comment]
Looks like it. 

#1953 might fix it.

**Comment by yurishkuro** [comment]
@pavolloffay I think this is fixed, but it seems that in the main repo in Docker Hub the `latest` tag is applied to the latest commit on master. This feels rather dangerous, we already have snapshots for unreleased builds, I would expect in the main repos the `latest` to refer to the latest official release, i.e. `1.15.1` right now.

**Comment by pavolloffay** [comment]
Note that before we were publishing `latest` as really the latest code in master. The snapshot repository does not have latest tag, one has to always use the commit SHA as the tag.

People should not depend in `latest` anyway, but it is a great way to share latest changes if one wants to use them.

**Comment by yurishkuro** [comment]
My proposal is to have `latest` in both repos:
  * in the main repo it points to the latest released version
  * in the snapshop repo it points to the latest commit on master

This provides both better safety and flexibility.

**Comment by pavolloffay** [comment]
Sounds fine to me, although to be consistent with the operator repo I would prefer to have `master` tag as part of the main repository. It also easier for users to change. 

The main motivation is also to get people using latest features before they are released to have more stable releases. Using a nonstandard tag name might cause less adoption.

We should also document this in https://www.jaegertracing.io/download/.

**Comment by asuffield** [comment]
I will just note that this is now failing again, the travis builds are reporting:
```
+docker push jaegertracing/jaeger-agent
The push refers to repository [docker.io/jaegertracing/jaeger-agent]
a40bc8102dba: Preparing
ce821040ffa8: Preparing
denied: requested access to the resource is denied
```

**Comment by jpkrohling** [comment]
Fixed by #2419.
